src/osd/PrimaryLogPG.cc

```diff
@@ -7053,6 +7077,134 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 
       break;
 
+    case CEPH_OSD_OP_CHUNK_WRITE_OR_GET:
```

Comment:
This logic looks right, but I think this should live in the same class as the GET/PUT.

In fact, now that I think about it, since we already have a specialized GET/PUT method (for chunks) that is different than the regular refcount behavior, perhaps we should create a new class calls 'cas' that collects all of these methods for managing the objects in the cas pool: WRITE_OR_GET, PUT, and eventually things like enumerating backreferences to support scrub, etc.  That will also let us collect the user interface into a cls_cas_client.h (or whatever) for non-tiering users of the CAS pool (for example, we expect that RGW will write chunks directly into the  cas tier).

Reply:
@liewegas I agree. A new class call is right way. I'll rebase this commit.